models-info/src/main/java/com/flipkart/fdp/ml/transformer/VectorAssemblerTransformer.java

```diff
@@ -20,27 +20,44 @@ public VectorAssemblerTransformer(final VectorAssemblerModelInfo modelInfo) {
     }
 
     private double[] predict (Object[] inputs) {
-        List<Double> output = new ArrayList<>();
+
+        int vectorSize = Stream.of(inputs)
+                .filter(o->isTypeDouble(o) || isTypeDoubleArray(o))
+                .mapToInt(o -> isTypeDoubleArray(o) ? ((double[])o).length : 1)
+                .sum();
+
+        double[] output = new double[vectorSize];
+        int i=0;
         for (Object input : inputs) {
             if (input == null) {
                 throw new RuntimeException("Values to assemble cannot be null");
             }
-            else if (double.class.equals(input.getClass()) ||
-                        Double.class.equals(input.getClass())) {
-                output.add((double)input);
+            else if (isTypeDouble(input)) {
+                output[i++] = (double) input;
             }
-            else if (double[].class.equals(input.getClass())) {
-                for (Double val : (double[])input) {
-                    output.add(val);
+            else if (isTypeDoubleArray(input)) {
+                for (double val : (double[])input) {
+                    output[i++] = val;
                 }
             }
             else {
                 throw new RuntimeException("Values to assemble cannot be of type: " + input.getClass().getCanonicalName());
             }
         }
-        return output.stream().mapToDouble(Double::doubleValue).toArray();
+        return output;
+    }
+
+    private boolean isTypeDouble(Object o) {
```

Comment:
I am ok with the current implementation of having a separate method if it adds to the readability/reuse. I will close this pull request.